account-kit/infra/src/alchemyTransport.ts

```diff
@@ -0,0 +1,171 @@
+  },
+  EthereumProvider["request"]
+>;
+
+export type AlchemyTransport = AlchemyTransportBase & {
+  chain: Chain;
+  updateHeaders(newHeaders: HeadersInit): void;
+};
+
+/**
+ * Creates an Alchemy transport with the specified configuration options.
+ * When sending all traffic to Alchemy, you must pass in one of rpcUrl, apiKey, or jwt.
+ * If you want to send Bundler and Paymaster traffic to Alchemy and Node traffic to a different RPC, you must pass in alchemyConnection and nodeRpcUrl.
+ *
+ * @example
+ * ### Basic Example
+ * If the chain you're using is supported for both Bundler and Node RPCs, then you can do the following:
+ * ```ts
+ * import { alchemy, sepolia } from "@account-kit/infra";
+ *
+ * const transport = alchemy({
+ *  // NOTE: you can also pass in an rpcUrl or jwt here or rpcUrl and jwt
+ *  apiKey: "your-api-key",
+ *  chain: sepolia,
+ * });
+ * ```
+ *
+ * ### AA Only Chains
+ * For AA-only chains, you need to specify the alchemyConnection and nodeRpcUrl since Alchemy only
+ * handles the Bundler and Paymaster RPCs for these chains.
+ * ```ts
+ * import { alchemy, zora } from "@account-kit/infra";
+ *
+ * const transport = alchemy({
+ *  chain: zora,
+ *  alchemyConnection: {
+ *    apiKey: "your-api-key",
+ *  },
+ *  nodeRpcUrl: "https://zora.rpc.url",
+ * });
+ * ```
+ *
+ * @param {AlchemyTransportConfig} config The configuration object for the Alchemy transport.
+ * @param {object} config.chain The blockchain network configuration.
+ * @param {number} config.retryDelay Optional The delay between retries, in milliseconds.
+ * @param {number} config.retryCount Optional The number of retry attempts.
+ * @param {string} [config.alchemyConnection] Optional Alchemy connection configuration (if this is passed in, nodeRpcUrl is required).
+ * @param {string} [config.fetchOptions] Optional fetch options for HTTP requests.
+ * @param {string} [config.nodeRpcUrl] Optional RPC URL for node (if this is passed in, alchemyConnection is required).
+ * @param {string} [config.rpcUrl] Optional RPC URL.
+ * @param {string} [config.apiKey] Optional API key for Alchemy.
+ * @param {string} [config.jwt] Optional JSON Web Token for authorization.
+ * @returns {AlchemyTransport} The configured Alchemy transport object.
+ */
+export function alchemy(config: AlchemyTransportConfig): AlchemyTransport {
+  const { retryDelay, retryCount, chain: chain_ } = config;
+  const chain = AlchemyChainSchema.parse(chain_);
+  const fetchOptions = config.fetchOptions ?? {};
+
+  const connectionConfig = config.alchemyConnection ?? config;
+
+  const rpcUrl =
+    connectionConfig.rpcUrl == null
+      ? `${chain.rpcUrls.alchemy.http[0]}/${connectionConfig.apiKey ?? ""}`
+      : connectionConfig.rpcUrl;
+
+  // TODO: we probably should just log these headers during telemetry logging instead of doing this mutable header stuff
+  fetchOptions.headers = new Headers(fetchOptions.headers);
+  fetchOptions.headers.set("Alchemy-AA-Sdk-Version", VERSION);
```

Comment:
@dphilipson finally fixed the issues in the previous PR doing this: https://github.com/alchemyplatform/aa-sdk/pull/563#discussion_r1561857400